name.go

```diff
@@ -30,7 +30,7 @@ func (e ErrInvalidName) Error() string {
 // NOTE: This function designed to work with name components, after they have
 // been split on the period rune.
 func isRuneInvalidForFirstCharacter(r rune) bool {
-	return (r < 'A' || r > 'Z') && (r < 'a' || r > 'z') && r != '_'
+	return (r < 'A' || r > 'Z') && (r < 'a' || r > 'z') && r != '_' && r != '-'
```

Comment:
Do you have a reason or test for this change?

Reply:
Yes, it is because the logicType validated-string uses an underscore.